### PR TITLE
433 support writing string arrays to disk v2

### PR DESCRIPTION
**Pull request by hokiegeek2**

### CODE REVIEW

File: tests/io_test.py [hunk 1]
```diff
@@ -243,14 +242,14 @@ def testGetDataSets(self):
...
     def testSaveStringsDataset(self):
         # Create, save, and load Strings dataset
-        strings_array = ak.array(['testing string{}'.format(num) for num in list(range(1,11))])
+        strings_array = ak.array(['testing string{}'.format(num) for num in list(range(0,25))])
         strings_array.save('{}/strings-test'.format(IOTest.io_test_dir), dataset='strings')
         r_strings_array = ak.load('{}/strings-test'.format(IOTest.io_test_dir), 
                                   dataset='strings')
-        self.assertTrue((strings_array == r_strings_array).all())
+        
+        self.assertTrue((strings_array.to_ndarray().sort() == r_strings_array.to_ndarray().sort()))
```
Comment (reuster986) [inline tests/io_test.py:252]: I believe this should have a `.all()` on the end?

File: tests/io_test.py [hunk 2]
```diff
@@ -261,41 +260,45 @@ def testSaveStringsDataset(self):
...
-        r_strings_array = ak.load('{}/strings-test'.format(IOTest.io_test_dir), 
-                                  dataset='strings')
-        self.assertTrue((strings_array == r_strings_array).all())       
+        strings = ak.array(['testing a longer string{} to be written, loaded and appended'.\
+                                  format(num) for num in list(range(0,26))])
+        strings.save('{}/strings-test'.format(IOTest.io_test_dir), dataset='strings')
+        n_strings = strings.to_ndarray().sort()
+        r_strings = ak.load('{}/strings-test'.format(IOTest.io_test_dir), 
+                                  dataset='strings').to_ndarray().sort()
+        self.assertTrue((n_strings == r_strings))
```
Comment (reuster986) [inline tests/io_test.py:272]: Should use `np.sort()` and `.all()` (see above).

File: tests/io_test.py [hunk 3]
```diff
@@ -261,41 +260,45 @@ def testSaveStringsDataset(self):
...
                      '{}/multi-type-test'.format(IOTest.io_test_dir))
         r_mixed = ak.load_all('{}/multi-type-test'.format(IOTest.io_test_dir))
-        self.assertTrue((strings_array == r_mixed['m_strings']).all())
-        self.assertTrue((m_floats == r_mixed['m_floats']).all())
-        self.assertTrue((m_ints == r_mixed['m_ints']).all())
 
-        r_floats = ak.load('{}/multi-type-test'.format(IOTest.io_test_dir), 
-                           dataset='m_floats')
+        self.assertTrue((strings_array.to_ndarray().sort() == \
+                                           r_mixed['m_strings'].to_ndarray().sort()))
```
Comment (reuster986) [inline tests/io_test.py:285]: Same as above

File: tests/io_test.py (hunk 1 above)
Comment (reuster986) [reply tests/io_test.py:252]: Actually, `.sort()` is an in-place operation, so the return value is `None`, and this comparison will not function as intended. I think what you want is to call `np.sort(strings_array.to_ndarray())`, which will return a sorted copy of the array. Then you can compare with `(a == b).all()`

File: tests/io_test.py [hunk 4]
```diff
@@ -261,41 +260,45 @@ def testSaveStringsDataset(self):
...
+                           dataset='m_floats'))
         self.assertTrue((m_floats == r_floats).all())
+
+        r_ints = ak.sort(ak.load('{}/multi-type-test'.format(IOTest.io_test_dir), 
+                           dataset='m_ints'))
+        self.assertTrue((m_ints == r_ints).all())
         r_strings = ak.load('{}/multi-type-test'.format(IOTest.io_test_dir), 
                             dataset='m_strings')
-        self.assertTrue((strings_array == r_strings).all())
+        self.assertTrue((strings_array.to_ndarray().sort() == r_strings.to_ndarray().sort()))
```
Comment (reuster986) [inline tests/io_test.py:298]: Same as above

File: tests/io_test.py [hunk 5]
```diff
@@ -318,10 +320,19 @@ def testAppendMixedStringsDataset(self):
...
+ 
+        r_floats = ak.sort(ak.load('{}/append-multi-type-test'.format(IOTest.io_test_dir), 
+                                   dataset='m_floats'))
+        r_ints = ak.sort(ak.load('{}/append-multi-type-test'.format(IOTest.io_test_dir), 
+                                 dataset='m_ints'))
+        self.assertTrue((m_floats == r_floats).all())
+        self.assertTrue((m_ints == r_ints).all())
+        
+        self.assertTrue((strings_array.to_ndarray().sort() == \
+                                                r_mixed['m_strings'].to_ndarray().sort()))
```
Comment (reuster986) [inline tests/io_test.py:335]: Here too

**Review by reuster986 — CHANGES_REQUESTED** [review]
@hokiegeek2 I think the changes to the Chapel src look good, but there are some issues with the test logic to fix before we trust the CI results.

File: tests/io_test.py (hunk 4 above)
Comment (hokiegeek2) [reply tests/io_test.py]: Fixed

File: tests/io_test.py (hunk 5 above)
Comment (hokiegeek2) [reply tests/io_test.py]: Fixed

File: tests/io_test.py (hunk 3 above)
Comment (hokiegeek2) [reply tests/io_test.py:293]: Fixed

File: tests/io_test.py (hunk 2 above)
Comment (hokiegeek2) [reply tests/io_test.py]: Fixed

File: tests/io_test.py (hunk 1 above)
Comment (hokiegeek2) [reply tests/io_test.py]: @reuster986 Argh, yeah, fixed in latest version I just pushed to github.

**Review by reuster986 — APPROVED** [review]
Changes look good!

**Comment by reuster986** [comment]
@hokiegeek2 git is saying there are conflicts in the branch, and it won't run the CI checks. Any idea what's going on?

**Comment by hokiegeek2** [comment]
> @hokiegeek2 git is saying there are conflicts in the branch, and it won't run the CI checks. Any idea what's going on?

Hey @reuster986, yeah, there's a conflict, will fix  in the AM.

**Comment by hokiegeek2** [comment]
@reuster986 it's ready to go:

image

**Pull request merged**